core/trino-main/src/main/java/io/trino/operator/table/Sequence.java

```diff
@@ -0,0 +1,301 @@
+                                    .build()),
+                    new DescribedTable(descriptor(ImmutableList.of("sequential_number"), ImmutableList.of(BIGINT))));
+        }
+
+        @Override
+        public TableFunctionAnalysis analyze(ConnectorSession session, ConnectorTransactionHandle transaction, Map<String, Argument> arguments)
+        {
+            Object startValue = ((ScalarArgument) arguments.get(START_ARGUMENT_NAME)).getValue();
+            if (startValue == null) {
+                throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "Start is null");
+            }
+
+            Object stopValue = ((ScalarArgument) arguments.get(STOP_ARGUMENT_NAME)).getValue();
+            if (stopValue == null) {
+                throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "Stop is null");
+            }
+
+            Object stepValue = ((ScalarArgument) arguments.get(STEP_ARGUMENT_NAME)).getValue();
+            if (stepValue == null) {
+                throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "Step is null");
+            }
+
+            long start = (long) startValue;
+            long stop = (long) stopValue;
+            long step = (long) stepValue;
+
+            if (start < stop && step <= 0) {
+                throw new TrinoException(INVALID_FUNCTION_ARGUMENT, format("Step must be positive for sequence [%s, %s]", start, stop));
+            }
+
+            if (start > stop && step >= 0) {
+                throw new TrinoException(INVALID_FUNCTION_ARGUMENT, format("Step must be negative for sequence [%s, %s]", start, stop));
+            }
+
+            return TableFunctionAnalysis.builder()
+                    .handle(new SequenceFunctionHandle(start, stop, start == stop ? 0 : step))
+                    .build();
+        }
+    }
+
+    public record SequenceFunctionHandle(long start, long stop, long step)
+            implements ConnectorTableFunctionHandle
+    {}
+
+    public static ConnectorSplitSource getSequenceFunctionSplitSource(SequenceFunctionHandle handle)
+    {
+        // using BigInteger to avoid long overflow since it's not in the main data processing loop
+        BigInteger start = BigInteger.valueOf(handle.start());
+        BigInteger stop = BigInteger.valueOf(handle.stop());
+        BigInteger step = BigInteger.valueOf(handle.step());
+
+        if (step.equals(BigInteger.ZERO)) {
+            checkArgument(start.equals(stop), "start is not equal to stop for step = 0");
+            return new FixedSplitSource(ImmutableList.of(new SequenceFunctionSplit(start.longValueExact(), stop.longValueExact())));
+        }
+
+        BigInteger splitJump = BigInteger.valueOf(DEFAULT_SPLIT_SIZE - 1).multiply(step);
+        ImmutableList.Builder<SequenceFunctionSplit> splits = ImmutableList.builder();
+        BigInteger splitStart = start;
+        do {
+            BigInteger splitStop = splitStart.add(splitJump);
+            if (pastStop(splitStop, stop, step)) {
```

Comment:
If you pre-align `stop` to a multiple of `step`, you don't need this check. All you have to do is pick the smallest (based on absolute value) between splitStop and stop.